src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.Cache.cs

```diff
@@ -172,17 +172,22 @@ public partial class JsonTypeInfo
             }
 
             // No cached item was found. Try the main dictionary which has all of the properties.
-            Debug.Assert(PropertyCache != null);
+#if DEBUG
+            if (PropertyCache == null)
```

Comment:
IIRC this can only happen during initialization races?

Reply:
I think so but I'm not sure how yet, hopefully the DebugInfo will shed some light on the "how"